Model/Resolver/EnvironmentVariableResolver.php

```diff
@@ -28,11 +28,11 @@ public function resolveValue($value)
     {
         $this->value = $value;
         return preg_replace_callback(
-            '/\{env:([^:\}\{]+?)\}/',
+            '/\%env\(([^:\%\%]+?)\)\%/',
```

Comment:
Just one consideration I'd like to share while seeing the changes (but I already thought about it yesterday with the previous syntax):

What do you think about to only allow common and portable environment parameter names.

I have something in mind like `A-Z` (uppercase only) and the underscore `_`. And a minimum length of three. Something along those lines just to keep problems out of the door. It would be good I could back this up with some good reference which I don't have at hand right now.

Perhaps also as this is PHP/HTTP context, also disallow any starting with `PHP_` and `HTTP_`.

You mileage may vary.

Thoughts?

Reply:
Why would you force people to have a specific env var schema? Let them name their stuff whatever they want I'd say. Not the responsibility of this extension, is it?

Reply:
Nah, not forcing the users, this is about portability and keeping some rough edges out. The behaviour can be declared undefined. Please share your use-case and your suggestion which would not make you feel forced into something. This would be helpful to me so that I can better understand your considerations and which kind of profile you have in your env to pass along configuration. From my book this is not to have NUL byte injections and such stuff, not to limit the user in what she's doing. Take care and great to see you joining the discussion, its been a long time! @DavidLambauer

Reply:
I do like this idea, I'll get one of my regex experts on it!

Reply:
Disallow list;

- `PHP_`
- `HTTP_`
- `SERVER_`
- `SCRIPT_`
- `QUERY_`
- `DOCUMENT_`

Reply:
Regex playground https://regex101.com/r/NcUNtB/2

Reply:
Since the commands of this extension are only executed in CLI environment, these env vars are very limited by itself and not directly exposed to third-party traffic. 

To be honest, I don't see much of a risk and not a need to restrict more. The configuration files are created manually by agencies/merchants/developers and the usually have control here. And if they explicitly choose to use such variables (maybe there's a reason, e.g. in multi-cluster setups), then the should be able to do this. 

However, I can understand the intention and  I'd be happy to merge a PR if you'd like to restrict that even further.